databrokerglobal/dxc
# feat : Developed version management system

To maintain different versions of DXC, developed a management system where following are key points
1. Docker Hub has "latest" tag 
2. WEB URL specifying version https://www.databroker.global/get_latest_dxc_version 
3. File .version containing current version. The value will be read and stored in database, when installed for first time or upgraded. Note, the file is deleted once done this process. When we upgrade then it is restored. 
4. New tab VERSION INFO is introduced to display current version and if there is any upgrade available
More info is documented in attached PDF file 
[DXC Versioning Strategy and Process.pdf](https://github.com/databrokerglobal/dxc/files/5828494/DXC.Versioning.Strategy.and.Process.pdf)

To test the fix, steps to follow
1. Install DXC. Make sure .version file is present and has same value as WEB URL
2. Click on VERSION INFO tab. displays version number and installed on date
3. Build new image with new version 
4. Change WEB URL value 
5. Click again on VERSION INFO tab. Message for upgrade is displayed
6. Restart DXC. It will ask for upgrade
7. Upgrade DXC (to simulate create a new .version file with new version number)  
8. Restart DXC. It will upgrade with new version
9. Check new version on tab
10. Success
